sqlmesh/core/config/connection.py

```diff
@@ -76,6 +77,11 @@ def _cursor_init(self) -> t.Optional[t.Callable[[t.Any], None]]:
         """A function that is called to initialize the cursor"""
         return None
 
+    @property
+    def is_recommended_for_state_sync(self) -> bool:
```

Comment:
This feels redundant. Can we just have a constant set with types of recommended engines?

Reply:
Addressed: https://github.com/TobikoData/sqlmesh/pull/2367/commits/ef2312fd98e2ddd139bb5b17bf11cfcdc98a4b4c